src/bsp/src/stm32/hal/decawave/platform/stm32f429zi/src/deca_port.c

```diff
@@ -0,0 +1,53 @@
+#include "deca_port.h"
+#include "main.h"
+#include "stm32f4xx_hal_conf.h"
+
+void deca_hardwareReset() {
+    GPIO_InitTypeDef GPIO_InitStruct;
+
+    // Enable GPIO used for DW1000 reset as open collector output
+    GPIO_InitStruct.Pin = DW_RESET_A_Pin;
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_OD;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(DW_RESET_A_GPIO_Port, &GPIO_InitStruct);
+
+    // drive the RSTn pin low
+    HAL_GPIO_WritePin(DW_RESET_A_GPIO_Port, DW_RESET_A_Pin, GPIO_PIN_RESET);
+
+    // Use HAL functions instead of FreeRTOS delays as this is called before the scheduler is
+    // started
+    HAL_Delay(1);
+
+    // put the pin back to tri-state ... as
+    // output open-drain (not active)
+    GPIO_InitStruct.Pin = DW_RESET_A_Pin;
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_OD;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_HIGH;
+    HAL_GPIO_Init(DW_RESET_A_GPIO_Port, &GPIO_InitStruct);
+    HAL_GPIO_WritePin(DW_RESET_A_GPIO_Port, DW_RESET_A_Pin, GPIO_PIN_SET);
+
+    HAL_Delay(100);
```

Comment:
Line 19 is the delay that the reset pin has to be driven low for for the DW to reset. (It could be in the range of a few microseconds, but the minimum we have with HAL_Delay() is 1 ms).

The delay on line 30 is the time needed for the DW to boot up. In the original code, this is more around the range of 10ms, but Phil had trouble with it, so he bumped it up for it to work properly. As this is called on initialisation of the board, a long delay isn't really an issue.

Reply:
Thanks for the explanation!